sound/soc/sof/intel/hda-dsp.c

```diff
@@ -428,6 +428,8 @@ static int hda_suspend(struct snd_sof_dev *sdev, bool runtime_suspend)
 		return ret;
 	}
 
+	sdev->dsp_power_state = SOF_DSP_D3;
+
```

Comment:
Better to set D3 in the core driver, make D0 and D3 be the more generic states?

Reply:
I think it makes sense here because the core requests the state and here's where the actual assignement happens isnt it?

Reply:
> I think it makes sense here because the core requests the state and here's where the actual assignement happens isnt it?

My point is that for platforms which don't have platform .suspend() ops, the dsp_power_state will not be changed then?

Reply:
Yes, it will not and thats correct right? You never change the DSP state for those platforms.